src/context/WalletProvider/NativeWallet/components/LegacyLogin.tsx

```diff
@@ -41,36 +36,70 @@ export const LegacyLogin = () => {
     handleSubmit,
     register,
     formState: { errors, isSubmitting },
+    reset,
   } = useForm({ shouldUnregister: true })
 
   const translate = useTranslate()
 
-  const isLoginError = (err: any): err is LoginError => err && typeof err.message === 'string'
+  const isLoginError = (err: any): err is LoginError =>
+    err && typeof err.response?.data?.error?.msg === 'string'
 
-  const onSubmit = async (values: FieldValues) => {
-    try {
-      // @TODO: Replace this with API call when backend is ready
-      const { mnemonic } = await Promise.resolve({ mnemonic: DUMMY_MNEMONIC })
+  const isDecryptionError = (err: any): err is DecryptionError =>
+    err &&
+    typeof err.message === 'string' &&
+    err.message?.startsWith('Native wallet decryption failed')
 
-      if (!bip39.validateMnemonic(mnemonic)) {
-        // @TODO: Should we send an error to the user?
-      }
+  type LoginResponse = {
+    success: boolean
+    data: string
+    error: any
+  }
 
-      // TODO: use response to create wallet vault.
+  const MIGRATE_URL = getConfig().REACT_APP_WALLET_MIGRATION_URL
+
+  const onSubmit = async (values: FieldValues) => {
+    try {
+      const hashedPassword = await getPasswordHash(values.email, values.password)
+      const { data: response } = await axios.post<LoginResponse>(`${MIGRATE_URL}`, {
+        email: values.email,
+        password: hashedPassword,
+        twoFactorCode: values.twoFactorCode || undefined,
+      })
+      const { data } = response
+      const encryptedWallet = data
       const vault = await Vault.create(undefined, false)
       vault.meta.set('createdAt', Date.now())
-      vault.set('#mnemonic', mnemonic)
+      vault.set(
+        '#mnemonic',
+        await decryptNativeWallet(values.email, values.password, encryptedWallet),
```

Comment:
@mrnerdhair I wanted to get your eyes on this to make sure this is the best way to decrypt the mnemonic.

Reply:
without going down a rabbit hole - the email and password should be treated with the same caution as a mnemonic - given they can be used to recovered the mnemonic.

Reply:
Once the vault is created we call reset() on the form to clear out the state.

Reply:
@mrnerdhair you good w/ this?

Reply:
Yeah, that's about as good as we can get it.